ui/raidboss/data/03-hw/dungeon/gubal_library_hard.js

```diff
@@ -0,0 +1,292 @@
+'use strict';
+
+// The Great Gubal Library--Hard
+[{
+  zoneRegex: /Great Gubal Library \(Hard\)/,
+  timelineFile: 'gubal_library_hard.txt',
+  timelineTriggers: [
+    {
+      id: 'Gubal Hard Triclip',
+      regex: /Triclip/,
+      beforeSeconds: 5,
+      condition: function(data) {
+        return data.role == 'healer' || data.role == 'tank';
+      },
+      infoText: {
+        en: 'Tank Buster',
+      },
+    },
+    {
+      id: 'Gubal Hard Searing Wind',
+      regex: /Searing Wind/,
+      beforeSeconds: 5,
+      infoText: function(data) {
+        if (data.role == 'tank') {
+          return {
+            en: 'Tank Cleave on you',
+          };
+        }
+        return {
+          en: 'Avoid tank cleave',
+        };
+      },
+    },
+    {
+      id: 'Gubal Hard Properties of Darkness',
+      regex: /Darkness \(buster\)/,
+      beforeSeconds: 5,
+      condition: function(data) {
+        return data.role == 'healer' || data.role == 'tank';
+      },
+      infoText: {
+        en: 'Tank Buster',
+      },
+    },
+  ],
+  triggers: [
+    {
+      id: 'Gubal Hard Bibliocide',
+      regex: / 14:1945:Liquid Flame starts using Bibliocide/,
+      condition: function(data) {
+        return data.role == 'healer';
+      },
+      infoText: {
+        en: 'AoE',
+      },
+    },
+    {
+      id: 'Gubal Hard Ferrofluid',
+      regex: / 1B:(\y{ObjectId}):(\y{Name}):....:....:(0030|0031):/,
+      condition: function(data, matches) {
+        return data.me == matches[2] || matches[1].slice(0, 1) == '4';
+      },
```

Comment:
\+ @JLGarber @quisquous 
Not to revive dead PRs or anything, but can either of you explain the above to me? This is the only instance in all of cactbot where we're matching on the ObjectId. It doesn't look to be erroneous since we're matching on the first character.

Reply:
[This was a deliberate choice to avoid translation issues](https://github.com/quisquous/cactbot/pull/693#discussion_r339323965). In [this revision](https://github.com/quisquous/cactbot/pull/693/commits/0d5e437dbce04c344fe7ac345ac10f6887eb08bf#diff-7bee985354c543175e0679847d231b98R68), we removed the hard-coded name for Liquid Flame, replacing it with a more generic test for "is this a non-player". This was the simplest(maybe?) solution we came up with to determine whether a particular overhead marker belonged to a player or the boss.

Reply:
Both? I guess I'm fine to work around it, but I just thought it was very odd and hadn't seen it before.

Reply:
Fair enough. Let me know if further updates to this trigger file would be helpful